apis/dns/v1alpha1/resource_record_set_types.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+)
+
+// ResourceRecordSetParameters define the desired state of a ResourceRecordSet
+type ResourceRecordSetParameters struct {
+	// Managed zone name that this ResourceRecordSet will be created in.
+	ManagedZone string `json:"managedZone"`
+
+	// Identifies what kind of resource this is.
+	//
+	// +kubebuilder:validation:Enum=dns#resourceRecordSet
+	Kind string `json:"kind"`
```

Comment:
Do we really need this as a parameter, given it is `required` and only option is `dns#resourceRecordSet` 🤔

Reply:
🤣 I was just being super purist. The GCP RRS API has this same field, and it can only be `dns#resourceRecordSet`. I didn't remove it because I was trying to reflect 100% the API. I will remove it!